src/apps/vpn/addons/addontutorial.cpp

```diff
@@ -120,6 +121,12 @@ void AddonTutorial::play(const QStringList& allowedItems) {
   m_activeTransaction = false;
 
   if (settingsRollbackNeeded()) {
+
+    MozillaVPN* vpn = MozillaVPN::instance();
```

Comment:
This PR creates a strong connection between the tutorial engine and the controller. Instead, i would suggest using signals:
1. Creates signals when a transaction begins, finalizes, rollbacks.
2. In the controller, connect that signals to "something" that does the correct logic.
Would it work?